http/events.go

```diff
@@ -298,6 +298,12 @@ func prependNamespacePatterns(patterns []string, requestNamespace *namespace.Nam
 // validateSubscribeAccessLoop continually checks if the request has access to the subscribe endpoint in
 // its namespace. If the access check ever fails, then the cancel function is called and  the function returns.
 func validateSubscribeAccessLoop(core *vault.Core, ctx context.Context, cancel context.CancelFunc, req *logical.Request) {
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
I'm concerned about this masking other problems.  I'd prefer to find a way to prevent the panic in the first place.  Thinking about right way to do that.

Reply:
All of the code in CheckToken I think assumes that we hold the read statelock.  We may have to do the same here.

Reply:
Hmm. The other places where `CheckToken` is called appear to be from outside `*Core*, so I would assume they wouldn't have had the `core.stateLock` either?

Reply:
Yeah, so maybe before we call CheckToken, call core.stateLock.RLock, and then make sure core.activeContext isn't erroring?

Reply:
Updated.